<!DOCTYPE html>
<html lang=en>
           <head>
            <meta charset=UTF-8>
            <title>Announcement elPrep open source project</title>
            <link rel=stylesheet
                  href=https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css
                  integrity=sha384-TX8t27EcRE3e/ihU7zmQxVncDAy5uIKz4rEkgIXeMed4M0jlfIDPvg6uqKI2xXr2
                  crossorigin=anonymous>
            <style>
section.tree {
    padding-left: 2em;
}
section.tree:first-child {
    padding-left: 0;
}
.article-link {
  margin-bottom: 1em;
}
</style>
<!-- Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-9X3G9MMWZP"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-9X3G9MMWZP');
</script>

           </head>
           <body>
            <header class="d-flex justify-content-center">
             <nav
                  class="navbar navbar-light bg-light w-100 mx-5 mb-3">
              <a class=navbar-brand href="/">Lisp HUG Maillist Archive</a>
             </nav>
            </header>
            <div class="d-flex justify-content-center">
             <div class="w-100 mx-5 px-3">
              <section class=tree>
               <article class=email>
                <h1>Announcement elPrep open source project</h1>
                <pre>
We have just released elPrep, a high-performance tool for preparing SAM/BAM/CRAM files for variant calling in DNA sequencing pipelines. It can be used as a replacement for standard tools, such as SAMtools and Picard, for preparation steps such as sorting, marking duplicates, reordering contigs, and so on, while producing identical results. elPrep is designed as a multi-threaded applications that runs entirely in memory, avoids repeated file I/O, and merges the computations of several preparation steps, to speed up the execution time by an order of magnitude. For example, on a 16-core server, we see a speedup of 10.5x when using elPrep compared to using a combination of SAMtools and Picard. elPrep is also a modular, extensible framework, where users can easily add more preparation steps that automatically take advantage of elPrepâ€™s inherent parallelism and performance.

elPrep has been developed at Imec Belgium at the ExaScience Life Lab (http://www.exascience.com), in collaboration with Intel and Janssen Pharmaceutica (Johnson &amp; Johnson). The application is implemented in Common Lisp, but relies on the symmetric multi-processing features from LispWorks.

The open source release is available at http://github.com/exascience/elprep with full end user documentation. A demo is available at http://github..com/exascience/elprep-demo including test data. The API documentation at http://exascience.github.io/elprep/elprep-package/ provides details about the elPrep framework.

---
Charlotte Herzeel, PhD
Researcher at ExaScience Life Lab (via IMEC)
Address: Kapeldreef 75, 3001 Leuven, Belgium




_______________________________________________
Lisp Hug - the mailing list for LispWorks users
lisp-hug@lispworks.com
http://www.lispworks.com/support/lisp-hug.html


</pre>
               </article>
              </section>
             </div>
            </div>
            <footer class="d-flex justify-content-center">
             <div>
              Updated at: 2020-12-07 08:29 UTC
             </div>
            </footer>
           </body>
          </html>